ios/Exponent/Kernel/Views/EXAppViewController.m

```diff
@@ -315,6 +318,34 @@ - (void)errorViewDidSelectRetry:(EXErrorView *)errorView
 
 - (UIInterfaceOrientationMask)supportedInterfaceOrientations
 {
+  NSString *validatedVersion = [[EXVersions sharedInstance] availableSdkVersionForManifest:_appRecord.appLoader.manifest];
+  NSString *majorSDKVersion = [validatedVersion componentsSeparatedByString:@"."][0];
+  if ([validatedVersion length] != 0 && [majorSDKVersion integerValue] <= 35) {
+    return [self legacySupportedInterfaceOrientations];
+  }
+  
+  if (_appRecord.experienceId == nil) {
+    return UIInterfaceOrientationMaskAllButUpsideDown;
+  }
+  EXScreenOrientationRegistry *registry = (EXScreenOrientationRegistry *)[UMModuleRegistryProvider getSingletonModuleForClass:[EXScreenOrientationRegistry class]];
+  if ([registry doesKeyExistForAppId:_appRecord.experienceId]) {
+    return [registry orientationMaskForAppId:_appRecord.experienceId];
+  } else if (_appRecord.appLoader.manifest) {
+    NSString *orientationConfig = _appRecord.appLoader.manifest[@"orientation"];
+    UIInterfaceOrientationMask mask = UIInterfaceOrientationMaskAllButUpsideDown;
+    if ([orientationConfig isEqualToString:@"portrait"]) {
+      mask = UIInterfaceOrientationMaskPortrait;
+    } else if ([orientationConfig isEqualToString:@"landscape"]) {
+      mask = UIInterfaceOrientationMaskLandscape;
+    }
+    [registry setOrientationMask:mask forAppId:_appRecord.experienceId];
```

Comment:
Should we really set the mask in the registry? Let's maybe not? This way we have a clean definition that _masks set in the registry have been set by JS code_ in opposition to _masks in the registry have either been set by JS or have been inferred from the manifest by `EXAppViewController`_

Reply:
Hmm I think you are right, it is not necessary.

Reply:
After I tested it a little, I think it is necessary because of the role that the registry has. If we remove adding this to registry, before we set the mask for the first time, every time the module will ask the registry about the mask it will give back AllButUpsideDown, but the phone will not rotate because EXAppViewController will ask the manifest for the mask and it will return e.g. Portrait, so it is wrong behavior and the easiest way is just to set the registry and treat is as a source of truth for all classes. What do you think?